asyncscope.md

```diff
@@ -152,32 +154,52 @@ struct async_scope {
     async_scope& operator=(const async_scope&) = delete;
     async_scope& operator=(async_scope&&) = delete;
 
+    auto nest(sender) const&->nested_sender<Values…>;
+
     auto spawn(sender) const&->void;
     auto spawn_future(sender) const&->future_sender<Values…>;
 
-    auto empty() const&->empty_sender;
+    auto on_empty() const&->empty_sender;
 
     auto get_stop_source() & ->stop_source&;
     auto get_stop_token() const& ->stop_token;
     auto request_stop() & ->void;
 };
 
+template<class... Ts>
+struct nested_sender; // see below;
 
 template<class... Ts>
 struct future_sender; // see below;
+
+struct empty_sender; // see below;
 ```
 
 ## Lifetime
 
 An `async_scope` object must outlive work that is spawned on it. It should be viewed as owning the storage for that work.
 The `async_scope` may be constructed in a local context, matching syntactic scope or the lifetime of surrounding algorithms.
-The destructor of the `async_scope` will terminate if there is outstanding work in the scope at destruction time, therefore `empty` **must** be called before the `async_scope` object is destroyed.
+The destructor of an `async_scope` will terminate if there is outstanding work in the scope at destruction time, therefore `on_empty` **must** be called and the returned `empty_sender` **must** have started and completed with `void`, before the `async_scope` object is destroyed.
+
+## nest
+
+`nest(sender) const&->nested_sender<Values…>;`
+
+returns a `nested_sender` that extends the lifetime of the `async_scope` that produced it. 
+
+A call to `nest()` is not expected to incur allocations.
```

Comment:
Can we guarantee this if we completely drop the returned sender?

I'm thinking of implementing `spawn()` in terms of `nest()`, and based on the description, a valid implementation would be:
```c++
void spawn(sender auto snd) {
     sender auto rsnd = nest(std::move(snd));
     (void) rsnd; // drop the result
}
```

Reply:
I'm not sure how we can guarantee the lack of allocations. Looking into the implementation of `spawn()`, the first thing we do is allocate space for for `operation_state` object. Then we try to start this operation.

The only place in which we can store the operation state is the returned object. In `async_scope` we don't have storage for nearly-infinite operations that might be spawned in the class.

But, operation states cannot be moved around. Once we start the operation, the memory location must not change. Thus, we cannot move the operation state into the returned object.

With this, I can't see how we can avoid a memory allocation.

---

The only solution I see is using a custom allocator.
I.e., if we are in a loop that spawns 1000 computations, then we can preallocate space for 1000 operation states in a custom allocator, and pass that custom allocator down into the `spawn()` call.

Reply:
The returned object is not started by `nest`. So I think Kirk's intent here is that the allocation is performed by the call to `connect` on the returned sender - so no allocation is performed for the same reason no allocation is performed when most senders are constructeed.

Is that interpretation correct, Kirk? 

That said, I am not yet sure I understand the purpose of nest.

Reply:
Which means that `spawn...()` does the allocation too, presumably? Does `spawn` at least need access to an allocator then? That could either be from the scope or from the spawn call itself.